extensions/quartz/deployment/src/main/java/org/apache/camel/quarkus/component/quartz/deployment/QuartzProcessor.java

```diff
@@ -44,4 +61,43 @@ NativeImageResourceBuildItem nativeImageResources() {
     ReflectiveClassBuildItem registerForReflection() {
         return new ReflectiveClassBuildItem(false, false, QUARTZ_JOB_CLASSES);
     }
+
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflectionWithMethods() {
+        return new ReflectiveClassBuildItem(true, false, QUARTZ_JOB_CLASSES_WITH_METHODS);
+    }
+
+    @BuildStep
+    void registerForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClasses,
+            CombinedIndexBuildItem combinedIndex, CurateOutcomeBuildItem curateOutcome) {
+        IndexView index = combinedIndex.getIndex();
+
+        ApplicationModel applicationModel = curateOutcome.getApplicationModel();
+        boolean oracleBlobIsPresent = applicationModel.getDependencies().stream()
+                .anyMatch(d -> d.getGroupId().equals("com.oracle.database.jdbc"));
+
+        final String[] delegatesImpl = index
+                .getAllKnownSubclasses(SQL_JDBC_DELEGATE)
+                .stream()
+                .map(c -> c.name().toString())
+                .filter(n -> oracleBlobIsPresent || !n.contains("oracle"))
+                .toArray(String[]::new);
+
+        reflectiveClasses.produce(new ReflectiveClassBuildItem(false, true, delegatesImpl));
+
+    }
+
+    @BuildStep
+    void indexSaxonHe(BuildProducer<IndexDependencyBuildItem> deps) {
+        deps.produce(new IndexDependencyBuildItem("org.quartz-scheduler", "quartz"));
+        deps.produce(new IndexDependencyBuildItem("org.apache.camel.quarkus", "camel-quarkus-quartz"));
```

Comment:
Do we need to index ourselves?

Reply:
I'm not sure, why it is behave that way, but:

- If index to itself is removed, registration for reflection (by annotation) does not work, and I'm getting `Caused by: java.lang.ClassNotFoundException: org.apache.camel.quarkus.component.quartz.CamelQuarkusQuartzConnectionProvider`
Therefore I kept index of ourselves

Reply:
> registration for reflection (by annotation) does not work

We should register it for reflection in the extension. No need to force that on to users.

Reply:
That is the fact where I'm not sure why it is wrong. I put `@RegisterForReflection` on `org.apache.camel.quarkus.component.quartz.CamelQuarkusQuartzConnectionProvider` in extensions/quartz/runtime module. I thought, that classes from the extensions are registered automatically. But it works only in case, that index contains own extension dependency. There is no need from the user to register it.

Reply:
I'm trying to register it in the extension (via annotation), I can try to register it via processor instead. I'll try it.

Reply:
Registration not via Annotation but bia BuildItem works, I'll fix it.